src/lib.jl

```diff
@@ -0,0 +1,10245 @@
+
+const time_t = __time_t
+
+const pthread_t = Culong
+
+struct __pthread_internal_list
+    __prev::Ptr{__pthread_internal_list}
+    __next::Ptr{__pthread_internal_list}
+end
+
+const __pthread_list_t = __pthread_internal_list
+
+struct pthread_mutex_t
+    data::NTuple{40, UInt8}
+end
+
+function Base.getproperty(x::Ptr{pthread_mutex_t}, f::Symbol)
+    f === :__data && return Ptr{__pthread_mutex_s}(x + 0)
+    f === :__size && return Ptr{NTuple{40, Cchar}}(x + 0)
+    f === :__align && return Ptr{Clong}(x + 0)
+    return getfield(x, f)
+end
+
+function Base.getproperty(x::pthread_mutex_t, f::Symbol)
+    r = Ref{pthread_mutex_t}(x)
+    ptr = Base.unsafe_convert(Ptr{pthread_mutex_t}, r)
+    fptr = getproperty(ptr, f)
+    GC.@preserve r unsafe_load(fptr)
+end
+
+function Base.setproperty!(x::Ptr{pthread_mutex_t}, f::Symbol, v)
+    unsafe_store!(getproperty(x, f), v)
+end
+
+struct pthread_cond_t
+    data::NTuple{48, UInt8}
+end
+
+function Base.getproperty(x::Ptr{pthread_cond_t}, f::Symbol)
+    f === :__data && return Ptr{__JL_Ctag_1115}(x + 0)
+    f === :__size && return Ptr{NTuple{48, Cchar}}(x + 0)
+    f === :__align && return Ptr{Clonglong}(x + 0)
+    return getfield(x, f)
+end
+
+function Base.getproperty(x::pthread_cond_t, f::Symbol)
+    r = Ref{pthread_cond_t}(x)
+    ptr = Base.unsafe_convert(Ptr{pthread_cond_t}, r)
+    fptr = getproperty(ptr, f)
+    GC.@preserve r unsafe_load(fptr)
+end
+
+function Base.setproperty!(x::Ptr{pthread_cond_t}, f::Symbol, v)
+    unsafe_store!(getproperty(x, f), v)
+end
+
+const pthread_once_t = Cint
+
+struct pthread_rwlock_t
+    data::NTuple{56, UInt8}
+end
+
+function Base.getproperty(x::Ptr{pthread_rwlock_t}, f::Symbol)
+    f === :__data && return Ptr{__JL_Ctag_1114}(x + 0)
+    f === :__size && return Ptr{NTuple{56, Cchar}}(x + 0)
+    f === :__align && return Ptr{Clong}(x + 0)
+    return getfield(x, f)
+end
+
+function Base.getproperty(x::pthread_rwlock_t, f::Symbol)
+    r = Ref{pthread_rwlock_t}(x)
+    ptr = Base.unsafe_convert(Ptr{pthread_rwlock_t}, r)
+    fptr = getproperty(ptr, f)
+    GC.@preserve r unsafe_load(fptr)
+end
+
+function Base.setproperty!(x::Ptr{pthread_rwlock_t}, f::Symbol, v)
+    unsafe_store!(getproperty(x, f), v)
+end
+
+struct tm
+    tm_sec::Cint
+    tm_min::Cint
+    tm_hour::Cint
+    tm_mday::Cint
+    tm_mon::Cint
+    tm_year::Cint
+    tm_wday::Cint
+    tm_yday::Cint
+    tm_isdst::Cint
+    tm_gmtoff::Clong
+    tm_zone::Ptr{Cchar}
+end
+
+struct aws_allocator
+    mem_acquire::Ptr{Cvoid}
+    mem_release::Ptr{Cvoid}
+    mem_realloc::Ptr{Cvoid}
+    mem_calloc::Ptr{Cvoid}
+    impl::Ptr{Cvoid}
+end
+
+"""
+    aws_allocator_is_valid(alloc)
+
+Inexpensive (constant time) check of data-structure invariants.
+
+### Prototype
+```c
+bool aws_allocator_is_valid(const struct aws_allocator *alloc);
+```
+"""
+function aws_allocator_is_valid(alloc)
+    ccall((:aws_allocator_is_valid, libaws_c_common), Bool, (Ptr{aws_allocator},), alloc)
```

Comment:
could we switch to the `@ccall` macro style? I've started using it and really like how it simplifies/minifies things